client_data/client_data/jobs.lua

```diff
@@ -13,7 +13,9 @@ return setmetatable({}, {
         }
     end,
     __pairs = function(t)
-        return function(t, k)
+        return function(t, i)
+            i = i + 1
+            if i < size then return i, t[i] end
```

Comment:
Nothing wrong with this, but to keep consistency with other libs let us go with:
```lua
if i == size then
    return nil, nil
end

return i, t[i]
```

We usually formulate it to check the abort condition and abort early, rather than the success condition and return conditionally. Also, we always return as many `nil`s when we abort as we return valid values on success. The cases where it matters are very few and far between, but when they appear it is nice to have.